src/Microsoft.CodeQuality.Analyzers/UnitTests/ApiDesignGuidelines/CollectionPropertiesShouldBeReadOnlyTests.cs

```diff
@@ -194,6 +207,48 @@ class A
 ");
         }
 
+        [Fact]
+        public void CSharp_CA2227_NonCollection()
+        {
+            VerifyCSharp(@"
+using System;
+
+public class A
+{
+    public string Name { get; set; }
+}
+");
+        }
+
+        [Fact, WorkItem(1900, "https://github.com/dotnet/roslyn-analyzers/issues/1900")]
+        public void CSharp_CA2227_ReadOnlyCollection()
```

Comment:
❗️ Should add tests for [`ReadOnlyCollection<T>`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.readonlycollection-1?view=netframework-4.7.2) and [`ReadOnlyDictionary<TKey, TValue>`](https://docs.microsoft.com/en-us/dotnet/api/system.collections.objectmodel.readonlydictionary-2?view=netframework-4.7.2) regardless of how we choose to handle them.